store/postgres/src/relational/prune.rs

```diff
@@ -495,10 +499,6 @@ impl Layout {
             catalog::set_last_pruned_block(conn, &self.site, &table.name, req.earliest_block)?;
         }
 
-        // Analyze the new tables
-        let tables = prunable_tables.iter().map(|(table, _)| *table).collect();
-        self.analyze_tables(conn, reporter, tables, cancel)?;
-
```

Comment:
With default settings, `Delete` will be used if we think we'll delete between 5% and 50% of the table, where the latter is a lot and statistics from before deletion will likely be junk. It would be better to analyze the tables in the `Delete` branch for which `catalog::needs_autoanalyze` says that an analyze is needed. That would basically cut out waiting for the autoanalyze daemon to get around to analyzing these tables.